lib/libc/Kconfig

```diff
@@ -68,6 +68,12 @@ config EXTERNAL_LIBC
 
 endchoice # LIBC_IMPLEMENTATION
 
+config PICOLIBC_SELECT_MODULE
+	bool "Use picolibc module"
+	default y
+	help
+	  Use picolibc module instead of picolibc included with toolchain
+
```

Comment:
I assume this commit message relates to those changes:
>  That creates a circular dependency if PICOLIBC_USE_MODULE is placed under the
picolibc configuration block.

I'm unsure exactly which circular dependency you refer to, but looking at the picolibc Kconfig I observe that there are now the following picolibc module related Kconfigs:
`ZEPHYR_PICOLIBC_MODULE`: Generated, promptless and `y` when the Picolibc module is available as Zephyr module
`PICOLIBC_MODULE`: Defined in the module, has a prompt, meaning it's user-configurable. https://github.com/zephyrproject-rtos/picolibc/blob/98e59b70d6ef8eb18f7dccc3d39cecaa0658fa4f/zephyr/Kconfig#L4
`PICOLIBC_USE_MODULE`: Defined in Zephyr, has a prompt, meaning it's user-configurable, seems to be a redundant setting of `PICOLIBC_MODULE`.

and now also
`PICOLIBC_SELECT_MODULE`

Why are all those needed ?
To me it seems they all try to achieve the same goal, provide the user with the option of enabling picolibc  as module.

This has some flaws, mainly that if user specifies `CONFIG_PICOLIBC_MODULE=y` then picolibc is built as a modules, but code below this line is never activated:
https://github.com/zephyrproject-rtos/zephyr/blob/6cbb3f5eece3bf63350a38d11a56e8b5923e5a2e/lib/libc/picolibc/Kconfig#L43

For that, the user must use `CONFIG_USE_PICOLIBC_MODULE=y`

The new `PICOLIBC_SELECT_MODULE` jump increases complexity.
It doesn't enable anything, afaict, it only changes the default behavior.
That means that if `PICOLIBC_SELECT_MODULE` is set to true **after** the initial run then nothing is changed.
A variant of the stuck-symbol.


I believe it should be possible to accomplish this using just a single symbol, like:
```
config PICOLIBC_MODULE
	bool "picolibc module"
	depends on ZEPHYR_PICOLIBC_MODULE && !CPLUSPLUS
	default y
	help
	  Use picolibc module instead of picolibc included with toolchain
```

This will ensure that the picolibc module can only be enabled if the module is available and C++ is not used.

You can then consider if the Kconfig symbol here should be kept (Kconfig allows the same symbol to be defined multiple times), if it should be made promptless, or even removed:
https://github.com/zephyrproject-rtos/picolibc/blob/98e59b70d6ef8eb18f7dccc3d39cecaa0658fa4f/zephyr/Kconfig#L4
can the